components/pkg-export-tar/src/build.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs as stdfs;
+#[cfg(target_os = "linux")]
+use std::os::unix::fs::symlink;
+use clap;
+use common;
+use common::command::package::install::InstallSource;
+use common::ui::{UI, Status};
+use tempdir::TempDir;
+use std::path::Path;
+use hcore::package::PackageIdent;
+use error::Result;
+use hcore::PROGRAM_NAME;
+use hcore::fs::{CACHE_ARTIFACT_PATH, CACHE_KEY_PATH, cache_artifact_path, cache_key_path};
+
+use super::{VERSION, BUSYBOX_IDENT};
+
+use rootfs;
+
+const DEFAULT_HAB_IDENT: &'static str = "core/hab";
+const DEFAULT_LAUNCHER_IDENT: &'static str = "core/hab-launcher";
+const DEFAULT_SUP_IDENT: &'static str = "core/hab-sup";
+
```

Comment:
It's probably worth adding some comments here about the intended future consolidation and finalization of the `BuildRoot` abstraction across exporters, so future devs have some more context on what otherwise looks like plain duplication of this code from the Docker exporter.

A similar note over in the Docker exporter (even if only to point to the comment here) could also be helpful, just to tie everything together.

Reply:
Agreed!  Just created an issue around refactoring this duplicate logic into it's own crate https://github.com/habitat-sh/habitat/issues/4522, will add some comment notes with links to this issue to both the tar and docker exporters.